p2p/handshake/handshake.go

```diff
@@ -52,18 +52,19 @@ type HandshakeAck struct {
 }
 
 // New instantiates handshake protocol for the host.
-func New(h host.Host, genesisID types.Hash20, opts ...Opt) *Handshake {
+func New(h host.Host, genesisID types.Hash20, prefix string, opts ...Opt) *Handshake {
 	ctx, cancel := context.WithCancel(context.Background())
 	hs := &Handshake{
 		logger:    log.NewNop(),
 		genesisID: genesisID,
+		protocol:  protocol.ID(fmt.Sprintf("%s/%s", prefix, hsprotocol)),
 		h:         h,
 		cancel:    cancel,
 	}
 	for _, opt := range opts {
 		opt(hs)
 	}
-	h.SetStreamHandler(protocol.ID(hsprotocol), hs.handler)
+	h.SetStreamHandler(hs.protocol, hs.handler)
```

Comment:
i reliazed that i don't know what happens if node connects, and it doesn't have any protocols in common (because of prefix), then what? will it remain using a connection or not?